site/topics/control-logic/control-logic.rst

```diff
@@ -15,13 +15,247 @@ Control Logic
+
+    The instruction register stores the operator and operand of the current instruction. The operand is output to the
+    bus for use during the execution of the instruction. The operator remains in the register for processing.
+
+
+* Consider what is stored in the instruction register and what is being output
+
+    * The operator, specifying some instruction to execute
+
+        * The most significant four bits in the instruction
+
+
+    * The operand, specifying some data for use in execution of the specific instruction
+
+        * The least significant four bits in the instruction
+
+
+* As previously discussed, each instruction requires several steps
+
+    * For example, loading data into register A from RAM (``LDAR/0001``)
+
+        * Output the operand from the instruction register and put it into the address register
+        * Output the value from the specified RAM address to the A register
+
+
+* Additionally, before any processing of any instruction can occur, it must be fetched from RAM
+
+    * No matter what instruction is to be executed, the two steps in the fetch cycle must happen
+
+
+* In other words, instructions require several steps
+
+
+Microcode Counter
+-----------------
+
+* The operator can be used to specify the steps to be executed in order to perform the instruction
+
+    * The operator would manipulate the control signals for the system
+
+
+* Thus, a look up table can be used to find the steps required to perform the full instruction
+
+    * Input to the look up table is the operand, output is the control signals
+
+
+* However, there is a problem with this idea since instruction may require several sequential steps too complete
+
+    * The microcodes
+
+
+* Consider the below table of the 4 microcode steps required for loading data from RAM to register A
+
+    * The first two steps are the fetch cycle
+    * The latter two steps move dara from the specified RAM address to register A
+
+
+.. list-table:: Full control logic of ``LDAR``
+    :widths: auto
+    :align: center
+    :header-rows: 1
+
+    * - :math:`Address`
+      - :math:`RAM`
+      - :math:`A`
+      - :math:`B`
+      - :math:`ALU`
+      - :math:`out`
+      - :math:`PC`
+      - :math:`Instruction`
+    * - ``1``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/1/0``
+      - ``0/0``
+    * - ``0``
+      - ``0/1``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0/1``
+      - ``1/0``
+    * - ``1``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0/0``
+      - ``0/1``
+    * - ``0``
+      - ``0/1``
+      - ``1/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0``
+      - ``0/0/0``
+      - ``0/0``
+
+
+* Not all instructions require 4 steps
+
+    * 1 instruction (``NOOP``) takes 2 microcode steps (fetch only)
+    * Six instructions take 3 steps
+    * Six instructions take 4 steps
```

Comment:
> This may be one of the more difficult ideas for the course to get down

but also, here's the answer with no lead in 🙈 

Not blocking, it's a theme in the course. This iteration lacks a "discovery" aspect for the material.